### PR TITLE
Adding implementation for the connector.

### PR DESCRIPTION
**Pull request by hemikak**
The following PR contains the required synapse mediators to make a call to an AS400 server program.

### CODE REVIEW

**Comment by hemikak** [comment]
The banners : 

mid-size-banner
big-banner
small-banner

File: src/main/java/org/wso2/carbon/connector/pcml/AS400CallProgram.java [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+            Resource pcmlFileResource =  registry.getResource(pcmlFileLocation);
+            InputStream pcmlFileContent = pcmlFileResource.getContentStream();
+
+            ProgramCallDocument pcmlDocument = new ProgramCallDocument(
+                    as400,
+                    FilenameUtils.getBaseName(pcmlFileLocation),
+                    pcmlFileContent,
+                    null,
+                    null,
+                    getFileType(FilenameUtils.getExtension(pcmlFileLocation)));
```
Comment (RKathees) [inline src/main/java/org/wso2/carbon/connector/pcml/AS400CallProgram.java]: Above lines can be in 2 or 3 lines. Please formate correctly.

File: src/main/java/org/wso2/carbon/connector/pcml/AS400Initialize.java [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+     * @param messageContext The message context.
+     * @param log            The logger object.
+     * @return The connection pool.
+     */
+    private AS400ConnectionPool getConnectionPool(MessageContext messageContext, SynapseLog log) {
+        AS400ConnectionPool as400ConnectionPool = null;
+        try {
+            Object poolNameParameter = getParameter(messageContext, AS400Constants.AS400_CONNECTION_POOL_NAME);
+            if (null != poolNameParameter) {
+
```
Comment (RKathees) [inline src/main/java/org/wso2/carbon/connector/pcml/AS400Initialize.java]: We don't need to put a blank space each line of codes. Put empty line to differentiate logics only. This should be fixed throughout this PR.

Comment (hemikak) [inline src/main/java/org/wso2/carbon/connector/pcml/AS400Initialize.java]: Fixed in https://github.com/wso2-extensions/esb-connector-pcml/pull/3/commits/22986634427adb228f51d47163e25fa13cbc10bc

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+                                    <outputDirectory>target/connector/dependencies</outputDirectory>
+                                </artifactItem>
+                                <!--<artifactItem>-->
+                                    <!--<groupId>net.sf.jt400</groupId>-->
+                                    <!--<artifactId>jt400</artifactId>-->
+                                    <!--<version>${jt400.version}</version>-->
+                                    <!--<type>jar</type>-->
+                                    <!--<overWrite>true</overWrite>-->
+                                    <!--<outputDirectory>target/connector/dependencies</outputDirectory>-->
+                                <!--</artifactItem>-->
```
Comment (hemikak) [inline pom.xml]: Removed in https://github.com/wso2-extensions/esb-connector-pcml/pull/3/commits/22986634427adb228f51d47163e25fa13cbc10bc

File: src/main/java/org/wso2/carbon/connector/pcml/AS400CallProgram.java [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+                throw new AS400PCMLConnectorException("A program name was not specified to call.");
+            }
+
+            // Get program name to call
+            String programName = (String) pcmlProgramNameParameter;
+
+            // Create program document by getting the PCML file from registry.
+            WSO2Registry registry = (WSO2Registry) messageContext.getConfiguration().getRegistry();
+            Resource pcmlFileResource =  registry.getResource(pcmlFileLocation);
+            InputStream pcmlFileContent = pcmlFileResource.getContentStream();
```
Comment (hemikak) [inline src/main/java/org/wso2/carbon/connector/pcml/AS400CallProgram.java]: Input stream close in https://github.com/wso2-extensions/esb-connector-pcml/pull/3/commits/22986634427adb228f51d47163e25fa13cbc10bc

File: src/main/java/org/wso2/carbon/connector/pcml/AS400CallProgram.java [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+
+/**
+ * A connector component that calls an AS400 program using PCML.
+ */
+public class AS400CallProgram extends AbstractConnector {
+
+    /**
+     * Input stream for the PCML file retrieved from registry.
+     */
+    private InputStream pcmlFileContent = null;
```
Comment (RKathees) [inline src/main/java/org/wso2/carbon/connector/pcml/AS400CallProgram.java]: pcmlFileContent can be declared as a local variable.

Comment (hemikak) [inline src/main/java/org/wso2/carbon/connector/pcml/AS400CallProgram.java]: Fixed in https://github.com/wso2-extensions/esb-connector-pcml/pull/3/commits/2fe192aa783f92fe930b1b4a6f100b3cbd558968

**Pull request merged**